vm/ubpf_vm.c

```diff
@@ -595,20 +599,61 @@ int
 ubpf_run(struct ubpf_vm *vm, void *mem, size_t mem_len, bool jit,
     uint64_t *bpf_return_value, char **errmsg)
 {
+    int cntrfd = 0;
+    ubpf_jit_fn fn;
+
+    if (vm->inst_cnt) {
+        cntrfd = setup_instruction_counter();
+        if (cntrfd == -1) {
+            vm->error_printf(stderr, "uBPF warning: unable to setup instruction counter\n");
+        }
+    }
+
     if (jit) {
-        ubpf_jit_fn fn = ubpf_compile(vm, errmsg);
+        if (vm->inst_cnt) {
+            enable_instruction_count(cntrfd);
+        }
+        fn = ubpf_compile(vm, errmsg);
+        if (vm->inst_cnt) {
+            disable_instruction_count(cntrfd);
+            vm->inst_cnt->inst_cmpl = get_instruction_count(cntrfd);
+            if (vm->inst_cnt->inst_cmpl < 0) {
```

Comment:
inst_cmpl is of type `size_t`, so if `get_instruction_count()` returns -1, it becomes a ULLONG_MAX, and we will never enter this if. I suggest we change the types of the fields in `struct inst_cnt` to long long as that's what get_instruction_count() returns, or just ssize_t.